packages/hardhat-core/src/internal/cli/cli.ts

```diff
@@ -41,14 +41,14 @@ async function printVersionMessage(packageJson: PackageJson) {
   console.log(packageJson.version);
 }
 
-function printWarningAboutNodeJsVersionIfNeceesary(packageJson: PackageJson) {
+function printWarningAboutNodeJsVersionIfNecessary(packageJson: PackageJson) {
   const requirement = packageJson.engines.node;
   if (!semver.satisfies(process.version, requirement)) {
     console.warn(
       chalk.yellow(
         `You are using a version of Node.js that is not supported by Hardhat, and it may work incorrectly, or not work at all.
 
-Please, upgrade your Node.js version.
+Please, make sure you are using a supported version of Node.js.
 
 To learn more about which versions of Node.js are supported go to https://hardhat.org/nodejs-versions`
```

Comment:
This isn't a comment on this PR per se, but I thought this was worth raising: I noticed that that link redirects to the higher-level heading "Stability Guarantees", but I wonder if it should instead go to the lower-level heading "Node.js versions support" (https://hardhat.org/reference/stability-guarantees.html#node-js-versions-support). I have no idea where those redirects are encoded... @alcuadrado ?

Reply:
I agree with this. Added a commit with a better redirect.